app/src/main/java/com/example/weather/repository/CityRepository.kt

```diff
@@ -4,20 +4,20 @@ import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import com.example.weather.appId
 import com.example.weather.cityIds
-import com.example.weather.network.CityDetails
-import com.example.weather.network.CityOverview
 import com.example.weather.network.CityRoomDatabase
+import com.example.weather.network.StateDetails
+import com.example.weather.network.StateOverview
 import com.example.weather.network.WeatherApi
 import com.example.weather.units
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 
 class CityRepository (private val database: CityRoomDatabase) {
 
-    private var _cityOverview = MutableLiveData<List<CityOverview>>()
-    val cityOverview: LiveData<List<CityOverview>> = _cityOverview
-    private var _cityDetails = MutableLiveData<CityDetails>()
-    val cityDetails: LiveData<CityDetails> = _cityDetails
+    private var _cityOverview= MutableLiveData<StateOverview>(StateOverview.Loading)
+    val cityOverview: LiveData<StateOverview> = _cityOverview
+    private var _cityDetails = MutableLiveData<StateDetails>(StateDetails.Loading)
+    val cityDetails: LiveData<StateDetails> = _cityDetails
 
     suspend fun refreshCityOverview() {
```

Comment:
the logic in this method needs some updating. right now, if there's an api error, it would just show the error screen instead of checking the cache first. try to take another look at this method and think about how you could rework it to handle every possible scenario